internal/cli/cli.go

```diff
@@ -48,10 +74,75 @@ func (c *RootCommand) persistentPreRunE(cmd *cobra.Command, args []string) error
 	return nil
 }
 
+func defaultServerOptions() []server.ServerFuncOpt {
+	userHeader := viper.GetString("grafana-user-header")
+
+	responseHeader := server.GrafanaResponseHeaders{
+		User: userHeader,
+	}
+
+	opts := []server.ServerFuncOpt{
+		server.WithCookieName(viper.GetString("cookie-name")),
+		server.WithGrafanaResponseHeaders(responseHeader),
+	}
+
+	return opts
+}
+
 func (c *RootCommand) runE(cmd *cobra.Command, args []string) error {
+
 	addr := viper.GetString("listen-address")
+	log.Infof("listening on %s", addr)
+
+	opts := defaultServerOptions()
+
+	adminUser := viper.GetString("admin-user")
+	adminPassword := viper.GetString("admin-password")
```

Comment:
nit: this can probably be combined  / moved to where BasicAuth is configured in the gapi.Config object.